pydra/engine/specs.py

```diff
@@ -39,19 +39,21 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
     @property
     def hash(self):
         """Compute a basic hash for any given set of fields."""
-        from .helpers import hash_function
+        from .helpers import hash_function, ensure_list
         from .helpers_file import hash_file
 
-        inp_dict = {
-            field.name: hash_file(getattr(self, field.name))
-            if (field.type == File and "container_path" not in field.metadata)
-            else getattr(self, field.name)
-            for field in attr_fields(self)
-            if (
-                field.name not in ["_graph_checksums", "bindings"]
-                and not field.metadata.get("output_file_template")
-            )
-        }
+        inp_dict = {}
+        for field in attr_fields(self):
+            if field.name not in [
+                "_graph_checksums",
+                "bindings",
+            ] and not field.metadata.get("output_file_template"):
+                if field.type == File and "container_path" not in field.metadata:
+                    inp_dict[field.name] = hash_file(getattr(self, field.name))
+                elif isinstance(getattr(self, field.name), tuple):
```

Comment:
just a note that we should address List/tuples of Files for hashing or dictionaries. i.e. nested input fields

Reply:
https://github.com/nipype/pydra/issues/178